src/server/scripts/Northrend/Naxxramas/boss_thaddius.cpp

```diff
@@ -288,88 +457,255 @@ class npc_stalagg : public CreatureScript
+                        me->SetReactState(REACT_AGGRESSIVE);
+                        me->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NOT_SELECTABLE);
+                        me->Talk(EMOTE_FEIGN_REVIVE, CHAT_MSG_MONSTER_EMOTE, 50000.0f, nullptr);
+                        isFeignDeath = false;
+                        
+                        refreshBeam = true; // force beam refresh
+
+                        if (Creature* feugen = ObjectAccessor::GetCreature(*me, instance->GetGuidData(DATA_FEUGEN)))
+                            if (feugen->GetVictim())
+                            {
+                                me->AddThreat(feugen->EnsureVictim(), 0.0f);
+                                me->SetInCombatWith(feugen->EnsureVictim());
+                            }
+                        break;
+                    case ACTION_TRANSITION:
+                        me->Kill(me); // true death
+                        me->DespawnOrUnsummon();
+
+                        if (Creature* coil = myCoil())
+                        {
+                            coil->CastStop();
+                            coil->Talk(EMOTE_TESLA_OVERLOAD, CHAT_MSG_RAID_BOSS_EMOTE, 50000.0f, nullptr);
+                        }
+                        break;
+                    case ACTION_TRANSITION_2:
+                        if (Creature* coil = myCoil())
+                            if (Creature* thaddius = ObjectAccessor::GetCreature(*me, instance->GetGuidData(DATA_THADDIUS)))
+                                coil->CastSpell(thaddius, SPELL_SHOCK_VISUAL);
+                    case ACTION_TRANSITION_3:
+                        if (GameObject* coil = myCoilGO())
+                            coil->SetGoState(GO_STATE_READY);
+                        break;
+                    default:
+                        break;
+                }
+            }
 
-            if (magneticPullTimer <= uiDiff)
+            void KilledUnit(Unit* victim) override
             {
-                if (Creature* pFeugen = ObjectAccessor::GetCreature(*me, instance->GetGuidData(DATA_FEUGEN)))
+                if (victim->GetTypeId() == TYPEID_PLAYER)
+                    Talk(SAY_STALAGG_SLAY);
+            }
+
+            void EnterCombat(Unit* who) override
+            {
+                Talk(SAY_STALAGG_AGGRO);
+
+                if (Creature* thaddius = ObjectAccessor::GetCreature(*me, instance->GetGuidData(DATA_THADDIUS)))
+                    thaddius->AI()->DoAction(ACTION_STALAGG_AGGRO);
+
+                if (Creature* feugen = ObjectAccessor::GetCreature(*me, instance->GetGuidData(DATA_FEUGEN)))
+                    if (!feugen->IsInCombat())
+                    {
+                        feugen->AddThreat(who, 0.0f);
+                        feugen->SetInCombatWith(who);
+                    }
+            }
+
+            void DamageTaken(Unit* /*who*/, uint32& damage) override
+            {
+                if (damage < me->GetHealth())
```

Comment:
Wouldn't it just make him "die" if damage is equal to his health (which is what's intended)? I may not be seeing it.